lib/src/rules/avoid_null_checks_in_equality_operators.dart

```diff
@@ -82,34 +84,48 @@ class AvoidNullChecksInEqualityOperators extends LintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    var visitor = _Visitor(this);
+    var visitor =
+        _Visitor(this, nnbdEnabled: context.isEnabled(Feature.non_nullable));
     registry.addMethodDeclaration(this, visitor);
   }
 }
 
 class _Visitor extends SimpleAstVisitor<void> {
   final LintRule rule;
+  final bool nnbdEnabled;
 
-  _Visitor(this.rule);
+  _Visitor(this.rule, {required this.nnbdEnabled});
 
   @override
   void visitMethodDeclaration(MethodDeclaration node) {
     var parameters = node.parameters?.parameters;
     if (parameters == null) {
       return;
     }
-    if (node.name.token.type == TokenType.EQ_EQ && parameters.length == 1) {
-      var parameter = DartTypeUtilities.getCanonicalElementFromIdentifier(
-          parameters.first.identifier);
-      bool checkIfParameterIsNull(AstNode node) =>
-          _isParameterWithQuestion(node, parameter) ||
-          (node is BinaryExpression &&
-              (_isParameterWithQuestionQuestion(node, parameter) ||
-                  _isComparingParameterWithNull(node, parameter)));
-
-      DartTypeUtilities.traverseNodesInDFS(node.body)
-          .where(checkIfParameterIsNull)
-          .forEach(rule.reportLint);
+
+    if (node.name.token.type != TokenType.EQ_EQ || parameters.length != 1) {
+      return;
+    }
+
+    var parameter = DartTypeUtilities.getCanonicalElementFromIdentifier(
+        parameters.first.identifier);
```

Comment:
I don't understand why we're not just using `parameter.first.declaredElement`. Is there some case that I'm missing?

Reply:
I'm not at all sure.  I notice a comment in `getCanonicalElement` with a bug reference that might be a hint?

```
        // A field element defined in a parameterized type where the values of
        // the type parameters are known.
        //
        // This concept should be invisible when comparing FieldElements, but a
        // bug in the analyzer causes FieldElements to not evaluate as
        // equivalent to equivalent FieldMembers. See
        // https://github.com/dart-lang/sdk/issues/35343.
```

It'd be worth tracking this down (but note that it's called in a bunch of places so we'd want to proceed with some extra caution).

Reply:
I added a question to the issue to double check, but I believe that `Element.declaration` should replace the need to check for `Member`s.